compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2801,7 +2801,13 @@ impl<'tcx> Ty<'tcx> {
     /// bound such as `[_]: Copy`. A function with such a bound obviously never
     /// can be called, but that doesn't mean it shouldn't typecheck. This is why
     /// this method doesn't return `Option<bool>`.
+    #[inline]
     pub fn is_trivially_sized(self, tcx: TyCtxt<'tcx>) -> bool {
+        #[inline(never)]
+        fn trivially_sized_cold_path<'tcx>(def: AdtDef<'tcx>, tcx: TyCtxt<'tcx>) -> bool {
+            def.sized_constraint(tcx).skip_binder().is_empty()
+        }
```

Comment:
(Looks at where this goes) oh wow this is expensive, it is a query. Shouldn't we just iterate over field types here, like we do for tuples?

Reply:
All of the other `is_trivially` just bail out for ADTs. I don't think `is_trivially` should ever fire a query. It's entire point is to avoid a full `Sized` query...

Reply:
If the ADT case just returns `false` every `no_core` program with an ADT requires a `Sized` lang item because codegen calls `Ty::is_sized`. I did not expect that. This breaks 26 rustdoc tests. I did not expect that.

Reply:
I also tried this:
```rust
ty::Adt(def, args) => def.variants().iter().all(|variant| {
    variant
        .tail_opt()
        .map(|tail| tail.ty(tcx, args).is_trivially_sized(tcx))
        .unwrap_or(true)
}),
```
Which is not fast because that `.ty(` calls the `type_of` query.

Reply:
I think it's completely okay to have `no_core`  require a `Sized` lang item.